src/components/AssetsTable/index.jsx

```diff
@@ -88,6 +95,33 @@ export class AssetsTable extends React.Component {
     });
   }
 
+  onLockClick(asset) {
+    this.props.toggleLockAsset(asset, this.props.courseDetails);
+  }
+
+  getImageThumbnail(thumbnail) {
+    const baseUrl = this.props.courseDetails.base_url ? this.props.courseDetails.base_url : '';
+    return thumbnail ? (<img src={`${baseUrl}${thumbnail}`} alt="Description not available" />) : 'Preview not available';
+  }
+
+  getLockButton(asset) {
+    const classes = [FontAwesomeStyles.fa];
+    let lockState;
+    if (asset.locked) {
+      lockState = 'Locked';
+      classes.push(FontAwesomeStyles['fa-lock']);
+    } else {
+      lockState = 'Unlocked';
+      classes.push(FontAwesomeStyles['fa-unlock']);
+    }
+    return (<Button
+      label={(<span className={classNames(...classes)} />)}
+      buttonType={'light'}
+      aria-label={`${lockState} ${asset.display_name}`}
+      onClick={() => { this.onLockClick(asset); }}
```

Comment:
I think this will create a new function instance when React compares component trees meaning that this will always trigger a re-render.

Check out [this article](https://github.com/vasanthk/react-bits/blob/master/gotchas/01.pure-render-checks.md#bad-2).

Here's [a more in-depth article](https://medium.com/@esamatti/react-js-pure-render-performance-anti-pattern-fb88c101332f)
> In Javascript different instances have different identities and thus the shallow equality check always produces false and tells React to re-render the components.

Reply:
Nice catch! Thanks!